### PR TITLE
Odom initializes only on first update in closed loop mode

### PR DESCRIPTION
**Pull request by Boberito25**

### CODE REVIEW

**Comment by Boberito25** [comment]
@pbaughman this appeared to work on a12...so i think its ready for QA

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 1]
```diff
@@ -290,6 +298,12 @@ namespace diff_drive_controller{
       left_pos  /= wheel_joints_size_;
       right_pos /= wheel_joints_size_;
 
+      if (odom_initialized)
```
Comment (po1) [inline diff_drive_controller/src/diff_drive_controller.cpp]: shouldn't this condition be reversed?

File: diff_drive_controller/src/diff_drive_controller.cpp [hunk 2]
```diff
@@ -290,6 +298,12 @@ namespace diff_drive_controller{
...
+      {
+        odometry_.init(time, left_pos, right_pos);
+        return;
```
Comment (po1) [inline diff_drive_controller/src/diff_drive_controller.cpp:305]: shouldn't we set `odom_initialized` to `true` when we are done?

**Comment by pbaughman** [comment]
Not sure if there's something special I need to do to test this, but it doesn't appear to be working.  I put ros_controllers/diff_drive_controller into a catkin workspace on a07, started him up, drove him down the hall, did sudo service gizmo restart and things were still way off center in rviz

**Review by po1 — APPROVED** [review]

**Comment by pbaughman** [comment]
I set this up in a workspace on a07 and I see the same problem - when I do sudo service gizmo restart after driving the robot away from it's starting place, the new starting place is super far away from the center.

**Comment by pbaughman** [comment]
PR is stale.  This is from over six months ago.

**Pull request closed without merging**